### PR TITLE
Fix socketpair failure when temporary directory has non-latin character in win32

### PR DESCRIPTION
**Pull request by nzh63**
Fixes: #1340

### CODE REVIEW

File: test/regress_buffer.c [hunk 1]
```diff
@@ -1332,7 +1333,7 @@ test_evbuffer_file_segment_add_cleanup_cb(void* ptr)
 	char const* arg = "token";
 
 	fd = regress_make_tmpfile("file_segment_test_file", 22, &tmpfilename);
-	tt_int_op(fd, >=, 0);
+	if (fd < 0) goto end;
```
Comment (azat) [inline test/regress_buffer.c:1336]: Why this has been changed?

File: test/regress_buffer.c [hunk 2]
```diff
@@ -1199,6 +1199,7 @@ test_evbuffer_add_file(void *ptr)
 	}
 
 	fd = regress_make_tmpfile(data, datalen, &tmpfilename);
+	if (fd < 0) goto end;
```
Comment (azat) [inline test/regress_buffer.c:1202]: Maybe better to use `tt_int_op` ?

File: test/regress_buffer.c [hunk 3]
```diff
@@ -2587,6 +2588,7 @@ test_evbuffer_freeze(void *ptr)
 	FREEZE_EQ(r, 0, -1);
 	len = strlen(tmpfilecontent);
 	fd = regress_make_tmpfile(tmpfilecontent, len, &tmpfilename);
+	if (fd < 0) goto end;
```
Comment (azat) [inline test/regress_buffer.c:2591]: `tt_int_op` ?

**Review by azat — APPROVED** [review]
Just a few minor comments in tests, apart from them, LGTM

File: test/regress_buffer.c (hunk 1 above)
Comment (nzh63) [reply test/regress_buffer.c:1336]: CTest saves and restores environment variables between tests, but it uses ANSI version `environ` instand of unicode version of `_wenviron`. Then, TMP environment variable will be corrupted and point to a invalid path,  so `regress_make_tmpfile` may fail. This is a issuse of cmake, not libevent.

I will create another merge requests in cmake to fix this issuse later. After ctext can handle unicode environment variables correctly, those lines will not need to be changed.

Or shall I remove the unicode test from ci?

Comment (azat) [reply test/regress_buffer.c:1336]: I like the test but, I don't understand but `tt_int_op(fd, >=, 0)` is almost the same as `if (fd < 0) goto end` (except for maybe additional logging and stuff), what did I miss?

Comment (nzh63) [reply test/regress_buffer.c:1336]: `tt_int_op` will mark the test case as failure by using `tinytest_set_test_failed_`, but ci require all test cases to be successful. I just use `goto end` to skip this test case.

Comment (azat) [reply test/regress_buffer.c:1336]: Yeah, forgot about this, indeed.

Okay, let's use `tt_skip()` (it is better then simply ignore the test) if the TMP environment variable is corrupted and we are on windows.
And also add a descriptive comment for this situation before calling `tt_skip`.

Comment (nzh63) [reply test/regress_buffer.c]: Sorry again. I change my TMP environment variable to asci-characters only days before, and those test cases was marks as OK, not SKIPPED. When I change TMP to include unicode, it become FAILED.

`tt_skip` is still indeed. I will chang it to `tt_skip`.

**Pull request merged**